core/nwb.misc.yaml

```
@@ -277,3 +277,33 @@ groups:
       value: volts
       doc: Unit of measurement. This value is fixed to 'volts'.
       required: false
+  - name: waveforms
+    neurodata_type_inc: VectorData
+    dtype: float
```

```suggestion
    dtype: float32
```
Best to be explicit about the minimum precision

---

I really don't like this. We are being "explicit" that the **minimum** precision is `float32`, which means this is allowed to be `float32`, or `float64`, in other words, any float. So I think putting `float` is much more clear.

---

The problem with `float` is that the user needs to know what dtype `float` maps onto. It is not obvious. In Python, `float` means float64, but here we mean float32 or float64. Similarly, with `int` (and `uint`), the user needs to know that `int` means int32 or int64 and NOT int8 or int16. So saying `int32` is clearer. By specifying dtype and number, the user needs to remember only that the number specifies the minimum precision, which they have to remember anyway, e.g., for cases where we use uint8.

---

We could make an exception for `float` where `float` == `float32` == float32 or float64, and `int` and `uint` would work differently, but I think that would be more confusing. We could remap `int` and `uint` to mean any int and any uint, but I think that would very confusing.

---

@rly You think it's confusing  to say "int means any type of integer and uint means any type of unsigned integer"? That seems pretty straightforward to me.